homeassistant/components/konnected/panel.py

```diff
@@ -0,0 +1,365 @@
+"""Support for Konnected devices."""
+import asyncio
+import logging
+
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    ATTR_STATE,
+    CONF_ACCESS_TOKEN,
+    CONF_BINARY_SENSORS,
+    CONF_DEVICES,
+    CONF_HOST,
+    CONF_ID,
+    CONF_NAME,
+    CONF_PIN,
+    CONF_PORT,
+    CONF_SENSORS,
+    CONF_SWITCHES,
+    CONF_TYPE,
+    CONF_ZONE,
+)
+from homeassistant.core import callback
+from homeassistant.helpers import aiohttp_client, device_registry as dr
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+from .const import (
+    CONF_ACTIVATION,
+    CONF_API_HOST,
+    CONF_BLINK,
+    CONF_DHT_SENSORS,
+    CONF_DISCOVERY,
+    CONF_DS18B20_SENSORS,
+    CONF_INVERSE,
+    CONF_MOMENTARY,
+    CONF_PAUSE,
+    CONF_POLL_INTERVAL,
+    CONF_REPEAT,
+    DOMAIN,
+    ENDPOINT_ROOT,
+    SIGNAL_SENSOR_UPDATE,
+    STATE_LOW,
+    ZONE_TO_PIN,
+)
+from .errors import CannotConnect
+
+_LOGGER = logging.getLogger(__name__)
+
+KONN_MODEL = "Konnected"
+KONN_MODEL_PRO = "Konnected Pro"
+
+# Indicate how each unit is controlled (pin or zone)
+KONN_API_VERSIONS = {
+    KONN_MODEL: CONF_PIN,
+    KONN_MODEL_PRO: CONF_ZONE,
+}
+
+
+class AlarmPanel:
+    """A representation of a Konnected alarm panel."""
+
+    def __init__(self, hass, config_entry):
+        """Initialize the Konnected device."""
+        self.hass = hass
+        self.config_entry = config_entry
+        self.config = (
+            config_entry.data
+        )  # a configuration.yaml device config contained in a device entry
+        self.host = self.config.get(CONF_HOST)
+        self.port = self.config.get(CONF_PORT)
+        self.client = None
+        self.status = None
+        self.api_version = KONN_API_VERSIONS[KONN_MODEL]
+
+    @property
+    def device_id(self):
+        """Device id is the MAC address as string with punctuation removed."""
+        return self.config.get(CONF_ID)
+
+    @property
+    def stored_configuration(self):
+        """Return the configuration stored in `hass.data` for this device."""
+        return self.hass.data[DOMAIN][CONF_DEVICES].get(self.device_id)
+
+    def format_zone(self, zone, other_items=None):
+        """Get zone or pin based dict based on the client type."""
+        payload = {
+            self.api_version: zone
+            if self.api_version == CONF_ZONE
+            else ZONE_TO_PIN[zone]
+        }
+        payload.update(other_items or {})
+        return payload
+
+    async def async_connect(self):
+        """Connect to and setup a Konnected device."""
+        try:
+            import konnected
```

Comment:
Please move the import to the top of the module.

Reply:
👍